### PR TITLE
Fix compatibility with pyuvdata v3.0

### PR DESCRIPTION
**Pull request by bhazelton**
## Description

Update warning handling for the upcoming pyuvdata v3.0.

## Types of changes

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Reference simulation update or replacement
- [ ] Documentation change (documentation changes only)
- [ ] Version change
- [ ] Build or continuous integration change

## Checklist:

For all pull requests:
- [x] I have read the [contribution guide](CONTRIBUTING.md).
- [x] My code follows the code style of this project.

Bug fix checklist:
- [ ] My fix includes a new test that breaks as a result of the bug (if possible).
- [x] All new and existing tests pass.
- [ ] I have checked that I reproduce the reference simulations or if there are differences they are explained below (if appropriate). If there are changes that are correct, I will update the reference simulation files after this PR is merged.
- [ ] I have checked (e.g., using the benchmarking tools) that this change does not significantly increase typical runtimes. If it does, I have included a justification in the comments on this PR.
- [ ] I have updated the [CHANGELOG](https://github.com/RadioAstronomySoftwareGroup/pyuvsim/blob/main/CHANGELOG.md).

### CODE REVIEW

File: src/pyuvsim/telescope.py [hunk 1]
```diff
@@ -134,12 +134,20 @@ def __init__(
...
-                        and not beam.future_array_shapes
-                    ):
-                        beam.use_future_array_shapes()
+                    with warnings.catch_warnings():
+                        warnings.filterwarnings(
+                            "ignore", "Future array shapes are now always used"
+                        )
+                        warnings.filterwarnings(
+                            "ignore",
+                            "The UVData.future_array_shapes attribute is now deprecated",
```
Comment (bhazelton) [inline src/pyuvsim/telescope.py:143]: This message is going to change, will need to update this.

File: src/pyuvsim/telescope.py [hunk 2]
```diff
@@ -438,13 +446,21 @@ def _str_to_obj(self, beam_id, beam_model, use_shared_mem=False):
...
-                # which was originally the only option.
-                uvb.read_beamfits(path, **read_kwargs)
-                warnings.warn(weird_beamfits_extension_warning, DeprecationWarning)
+            with warnings.catch_warnings():
+                warnings.filterwarnings(
+                    "ignore", "Future array shapes are now always used"
+                )
+                warnings.filterwarnings(
+                    "ignore",
+                    "The UVData.future_array_shapes attribute is now deprecated",
```
Comment (bhazelton) [inline src/pyuvsim/telescope.py:455]: This message is going to change, will need to update this.

File: src/pyuvsim/uvsim.py [hunk 3]
```diff
@@ -478,8 +478,13 @@ def uvdata_to_task_iter(
         raise TypeError("catalog must be a SkyModelData object.")
 
     # use future array shapes
-    if not input_uv.future_array_shapes:
-        input_uv.use_future_array_shapes()
+    with warnings.catch_warnings():
+        warnings.filterwarnings("ignore", "Future array shapes are now always used")
+        warnings.filterwarnings(
+            "ignore", "The UVData.future_array_shapes attribute is now deprecated"
```
Comment (bhazelton) [inline src/pyuvsim/uvsim.py:484]: This message is going to change, will need to update this.

File: src/pyuvsim/uvsim.py [hunk 4]
```diff
@@ -747,8 +752,13 @@ def run_uvdata_uvsim(
     ):
         raise ValueError("input_uv must have XX,YY,XY,YX polarization")
 
-    if not input_uv.future_array_shapes:
-        input_uv.use_future_array_shapes()
+    with warnings.catch_warnings():
+        warnings.filterwarnings("ignore", "Future array shapes are now always used")
+        warnings.filterwarnings(
+            "ignore", "The UVData.future_array_shapes attribute is now deprecated"
```
Comment (bhazelton) [inline src/pyuvsim/uvsim.py:758]: This message is going to change, will need to update this.

File: tests/test_uvsim.py [hunk 5]
```diff
@@ -115,7 +120,11 @@ def uvobj_beams_srcs():
     uv_obj, beam_list, beam_dict = pyuvsim.initialize_uvdata_from_params(
         param_dict, return_beams=True, force_beam_check=True
     )
-    assert uv_obj.future_array_shapes
+    with warnings.catch_warnings():
+        warnings.filterwarnings(
+            "ignore", "The UVData.future_array_shapes attribute is now deprecated"
```
Comment (bhazelton) [inline tests/test_uvsim.py:125]: This message is going to change, will need to update this.

**Review by mkolopanis — APPROVED** [review]

**Pull request merged**